### PR TITLE
Expand Pet Armors for Dogs [CR]

### PR DESCRIPTION
**Pull request by DracoGriffin**
#### Summary

```SUMMARY: Content "Adds new pet armors for canines with crafting recipes."```

#### Purpose of change

Adds new pet armors with focus on canines (with foundation of proper materials, can be expanded to other pets more easily) since merging of #28646.
#### Describe the solution

The materials listed for armors may not 100% reflect (but I kept them as close as possible) as balancing pet resilience (how many "Tough Zombies" they could kill one-on-one in a small lit room before dying) was very difficult as increasing material thickness by 1 could dramatically increase pet resilience.

For instance, the "Chainmail dog coat" when set to JUST
```
    "material": [ "iron" ],
    "material_thickness": 2,
```
would allow a German Shepherd to take down 21 Tough Zombies, but changing it to:
```
    "material": [ "iron", "leather" ],
    "material_thickness": 1,
```
and they could only take down 6.

Additionally, pet resilience is null against groups of zombies, they quickly die so keep in mind the following data when used just to balance one-on-one damage protection against average zombie attacks. (Interestingly, the amount of Tough Zombies the German Shepherds could kill was similar to the hits taken/amount of turns they lasted against Zombie Hulks, fun statistics!) 

Also, German Shepherds are rare and the strongest of the dog breeds (the average would be Labrador or Pit Bulls in both combat ability and group spawns), so resilience may be artificially higher than normal.

I followed the player armor trend to correlate pet armors (so there wouldn't be any abnormalities like Plastic giving more protection than Iron):

Naked/None <= Plastic < Leather <= Bone/Leather < Iron < Kevlar < Chitin < AcidChitin < Superalloy.

However, purely based on singular material did not uphold this so I had to force compliance by interweaving "similar" materials to follow this evolution more closely. This is what I ended up with:

_Average kills in parentheses._
Naked/None **(2)** <= Neoprene/Plastic **(4)** < Cotton/Leather **(7)** <= Bone/Leather **(8)** < Budget Steel/Iron **(14)** < Kevlar **(12)** < Chitin **(27)** < AcidChitin **(28)** < Superalloy **(infinite?)** .

Ideally I'd like to get to these numbers but would require further pet armor coding to allow greater control over cut/bash resistances, so is beyond this PR:
None **(2)** <= "Rainsuit" **(3)** < "Leather harness" **(6)** <= "Decorative bone leather harness" **(6-7)** < "Chainmail coat" **(9)** <= "Kevlar harness" **(9-10)** < "Chitin mesh harness" **(14)** < "AcidChitin mesh harness" **(16)** < "Superalloy outfit/harness" **(20)** .

Recipes were based on relevant armors and downsized somewhat appropriately by 25%, with some difficulties increased (shaping the material to an animal can be trickier given different locomotion [bipedal vs quadruped and gaits]).

[Inspiration for rainsuit](https://www.amazon.com/Muttluks-Four-Legged-Rainsuit-Size/dp/B00444YDYC).

[Inspiration for harnesses](https://www.dogsportgear.com/K9-Harness_c_283.html).

Added new crafting categories (since pet armor can be applied to different body types, can be added as they are made).

The superalloy version is basically locked to debug only, it was more for testing, but could be an extremely rare drop or perhaps made with the #27106 - Nanofabricator.

NO_SALVAGE applied to leather harness (gave back way too much rags and leather patches for some reason) and rainsuit (yielded neoprene sheets).

#### Describe alternatives you've considered

Keeping to mod only.

#### Additional context

~~Murder~~ Testing Room
Well-lit with overhead lights so Tough Zombie and (Armored) German Shepherd always had sight to engage; one Tough Zombie would be spawned on alternate staircases after each Tough Zombie died until German Shepherd perished. Corpses were tracked and cleaned up between tests (to avoid revivifications disrupting tests).
petarmortesting

### CODE REVIEW

**Comment by tinukedaya** [comment]
I can't imagine a dog being able to fight in chainmail armor. It would be more resilient, sure... but I would think also unable to fight effectively. I'd love to see some padded/leather/kevlar harnesses, yes. But that would be it. For dog to be effective in fighting it needs freedom of movement.

just my 2 cents.

**Comment by DracoGriffin** [comment]
> I can't imagine a dog being able to fight in chainmail armor. It would be more resilient, sure... but I would think also unable to fight effectively. I'd love to see some padded/leather/kevlar harnesses, yes. But that would be it. For dog to be effective in fighting it needs freedom of movement.
> 
> just my 2 cents.

Thank you for the feedback, I appreciate it.

Honestly, I am on board with that, as it has proven quite difficult to find some solid results from barding/armoring dogs. They have been [used throughout history](https://en.wikipedia.org/wiki/Dogs_in_warfare#History) and shown with differing types of armor, but I haven't found anything yet that states "was ineffective" or "improved survivability and kill ratios". Basically for chain mail, I can't determine how much was for combat and how much was for show/decorative.

Further reading: [Dogs of the Conquistadors](https://weaponsandwarfare.com/2019/01/12/dogs-of-the-conquistadors/), [Dogs of the Conquest](https://www.amazon.com/Dogs-Conquest-John-Grier-Varner/dp/0806117931)

Images: [1](https://sv.wikipedia.org/wiki/Stridshund#/media/File:Dog_Armour.jpg) [2](https://www.ripleys.com/weird-news/animal-armor/) [3](https://www.oldscrolls.com/shop_image/product/025603_5.jpg) [4](https://www.aliexpress.com/item/Anti-Bite-Chainmail-Stainless-Steel-Mesh-Protective-Dog-Suits-hunting-cloth-for-dog-cloth/32867905594.html)

The one thing that I didn't add because there isn't code to support it, but spiked collars/torsos seems consistent throughout history. So I could mimic "resilience" by not being able to attack or harm dog directly (because spikes hurt), but there wouldn't be any inflicted bleeding damage or damage at all.

>[Luath wore a heavy metal collar with sharpened spikes, and also hardened leather body armor to protect him from blows and knife wounds when he went into battle.](https://books.google.com/books?id=JhflH2NIXxUC&lpg=PA77&pg=PA23#v=onepage&q&f=false)

**Comment by tinukedaya** [comment]
What I was worried about is the kind of "full body" armors shown in some of the pictures. Those will hinder the attack abilities of the dogs in my opinion and were probably just decorative. 

But the more practical ones, like the of the anti-bite chain mail you pictured and those shown in the pics in the Dogs of the Conquistadors, those might work I guess. From what I can see, they have quite low coverage so still allow freedom of movement... So I will just think about it that way. Maybe have it somehow mentioned in the description? e.g. "A makeshift harness of chitin fitted to a thin mesh harness protecting the chest and shoulders of canines." 

Maybe sometime in the future the items would work with proper encumbrance and coverage and will affect the dogs stats. 

btw: the spike collars would be awesome!

File: data/json/items/armor/pets_dog_armor.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    "copy-from": "kevlar_harness",
+    "color": "brown",
+    "name": "leather dog harness",
+    "name_plural": "leather dog harnesses",
+    "description": "A harness made from leather that can be attached to a canine for protection.  You could put this on a friendly dog.",
+    "price": 17500,
+    "price_postapoc": 1750,
+    "material": [ "cotton", "leather" ],
+    "weight": 2145,
+    "flags": [ "NO_SALVAGE" ]
```
Comment (ZhilkinSerg) [inline data/json/items/armor/pets_dog_armor.json]: It is missing `IS_PET_ARMOR` flag.

Comment (DracoGriffin) [reply data/json/items/armor/pets_dog_armor.json]: Good catch, I forgot to re-check if this work after adding them (I quickly added them when I noticed butchering gave inappropriate amounts but assumed still worked). Also, totally forgot about using extend. Fixed!

**Comment by ZhilkinSerg** [comment]
Later we should make pet armor menu more like "Wear" menu:

image

image

**Comment by ZhilkinSerg** [comment]
AIso something for doing later - it would be nice if "Monster" and "Look Around" menu would show armor on pets:

image

image

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/latest-experimental-features/5582/1106

**Pull request merged**